api/server/router/swarm/cluster_routes.go

```diff
@@ -107,7 +107,12 @@ func (sr *swarmRouter) createService(ctx context.Context, w http.ResponseWriter,
 		return err
 	}
 
-	id, err := sr.backend.CreateService(service)
+	encodedAuth := ""
+	if auth, ok := r.Header["X-Registry-Auth"]; ok {
```

Comment:
@nishanttotla FYI actually we were wrong and @thaJeztah was right: `r.Header.Get("X-Registry-Auth")` is the way to go.